packages/thyra/core/src/interfaces/nonlinear/model_evaluator/fundamental/Thyra_ModelEvaluator.hpp

```diff
@@ -967,7 +1056,79 @@ class ModelEvaluator : public ModelEvaluatorBase {
    * </ul>
    */
   virtual RCP<LinearOpBase<Scalar> > create_DgDp_op( int j, int l ) const = 0;
-  
+
+  /** \brief If supported, create a linear operator derivative object for
+   * \f$\sum_k \lambda_k \frac{\partial^2 f_k}{\partial \boldsymbol{x}^2}\f$.
+   *
+   * <b>Preconditions:</b><ul>
+   * <li><tt>this->createOutArgs().supports(OUT_ARG_hess_f_xx)==true</tt>
+   * </ul>
+   */
+  virtual RCP<LinearOpBase<Scalar> > create_hess_f_xx() const { return Teuchos::null; };
```

Comment:
Should this not be per virtual and have the default implementation moved to the class `Thyra::ModelEvaluatorDefaultBase`?  This base class is supposed to be a pure virtual base class.